apis/v1alpha1/annotations.go

```diff
@@ -0,0 +1,10 @@
+package v1alpha1
+
+import "fmt"
+
+var (
+	// ResourceIDAnnotation is an annotation containing resource-identifying fields used for looking up API Gateway
+	// resources. The value of this annotation is used during resource lookups to detect any changes made to immutable,
+	// resource-identifying fields when updating a resource.
```

Comment:
Not sure I understand what is this needed for. The ID is stored in the status. Are you trying work around situations where the ID changes?

Reply:
> Are you trying work around situations where the ID changes?

Yeah, I was. But I found out that other controllers do not attempt to do this, so I have left it out and removed all changes related to that.